src/crush/CrushWrapper.h

```diff
@@ -1026,7 +1026,11 @@ class CrushWrapper {
 
   void start_choose_profile() {
     free(crush->choose_tries);
-    crush->choose_tries = (__u32 *)malloc(sizeof(*crush->choose_tries) * crush->choose_total_tries);
+    /*
+     * the original choose_total_tries value was off by one (it
+     * counted "retries" and not "tries").  add one to alloc.
+     */
+    crush->choose_tries = (__u32 *)malloc(sizeof(*crush->choose_tries) * (crush->choose_total_tries + 1));
```

Comment:
@tchaikov 
yes, I lost it. actually the original fix for me like your suggestion on on 5e0722f.
Maybe change memset() is enough and didn't change anything about mapper record.
I will update it.